### PR TITLE
Add blacklist functionality to http cache, fix max store size for http cache

### PR DESCRIPTION
**Pull request by davidhao3300**
and other random things

### CODE REVIEW

**Comment by davidhao3300** [comment]
@liuyang-li @jiangty-addepar

File: .gitignore [hunk 1]
```diff
@@ -16,6 +16,10 @@
 # Files created during tests
 /test/com/facebook/buck/testutil/integration/testlibs/buck-out/
 
+# IntelliJ
+*.iml
```
Comment (jiangty-addepar) [inline .gitignore:20]: I think buck purposefully checks in `.iml` files and perhaps things in `.idea`

File: .idea/modules.xml [hunk 2]
```diff
@@ -2,12 +2,31 @@
 <project version="4">
   <component name="ProjectModuleManager">
     <modules>
-      <module fileurl="file://$PROJECT_DIR$/third-party/java/aosp/aosp.iml" filepath="$PROJECT_DIR$/third-party/java/aosp/aosp.iml" />
-      <module fileurl="file://$PROJECT_DIR$/buck.iml" filepath="$PROJECT_DIR$/buck.iml" />
-      <module fileurl="file://$PROJECT_DIR$/tools/consistency_checker/consistency.iml" filepath="$PROJECT_DIR$/tools/consistency_checker/consistency.iml" />
-      <module fileurl="file://$PROJECT_DIR$/programs/programs.iml" filepath="$PROJECT_DIR$/programs/programs.iml" />
-      <module fileurl="file://$PROJECT_DIR$/python-dsl/python-dsl.iml" filepath="$PROJECT_DIR$/python-dsl/python-dsl.iml" />
-      <module fileurl="file://$PROJECT_DIR$/scripts/scripts.iml" filepath="$PROJECT_DIR$/scripts/scripts.iml" />
+      <module fileurl="file://$PROJECT_DIR$/project_root.iml" filepath="$PROJECT_DIR$/project_root.iml" />
```
Comment (jiangty-addepar) [inline .idea/modules.xml:5]: revert?

File: src/com/facebook/buck/android/AbstractAndroidGraphEnhancementResult.java [hunk 3]
```diff
@@ -59,7 +59,7 @@
 
   @Value.Derived
   default Optional<PreDexMerge> getPreDexMerge() {
-    return getDexMergeRule().transform(left -> Optional.of(left), right -> Optional.empty());
+    return getDexMergeRule().transform(left -> Optional.of(left), right -> Optional.<PreDexMerge>empty());
```
Comment (jiangty-addepar) [inline src/com/facebook/buck/android/AbstractAndroidGraphEnhancementResult.java:62]: revert?

**Review by jiangty-addepar — COMMENTED** [review]
Just wondering, what's our use case for these new features?

File: src/com/facebook/buck/android/AbstractAndroidGraphEnhancementResult.java (hunk 3 above)
Comment (davidhao3300) [reply src/com/facebook/buck/android/AbstractAndroidGraphEnhancementResult.java:62]: this fixes that compile error you noted in a buck issue a while back, do we not want it?

File: .idea/modules.xml (hunk 2 above)
Comment (davidhao3300) [reply .idea/modules.xml:5]: these were the intellij change i had to make to get things to work in intellij, are they unwanted?

File: .gitignore (hunk 1 above)
Comment (davidhao3300) [reply .gitignore:20]: this will ignore changes to any .idea files in the future. I don't believe they check in iml files

**Comment by davidhao3300** [comment]
The main usecase is the blacklisting stuff, the others are mainly things to make intellij work

File: .gitignore (hunk 1 above)
Comment (jiangty-addepar) [reply .gitignore:20]: they definitely check in at least one `.iml` file (at the root level); buck doesn't use `buck project` for its project files

File: .idea/modules.xml (hunk 2 above)
Comment (jiangty-addepar) [reply .idea/modules.xml:5]: I wouldn't suggest changing this; you shouldn't need to run `buck project` in buck

File: src/com/facebook/buck/android/AbstractAndroidGraphEnhancementResult.java (hunk 3 above)
Comment (jiangty-addepar) [reply src/com/facebook/buck/android/AbstractAndroidGraphEnhancementResult.java:62]: put it in a separate change I guess

**Comment by jiangty-addepar** [comment]
I guess what I'm asking is:
- What are we blacklisting / why do we need to blacklist them?
- Why do we need a max http cache size?

**Comment by epkugelmass** [comment]
Very unlikely that they will upstream changes to dot files.

**Comment by davidhao3300** [comment]
we can remove the changes to dot files when we open PRs to buck

blacklisting - we blacklist //lib... targets because we would download them every time otherwise (at least in jenkins, where we git clean)

max http cache size was something i tried out before having to implement blacklisting. I can remove that change for our fork, but it's something that should be merged upstream

**Comment by davidhao3300** [comment]
🥒

**Pull request closed without merging**